conda/trust/signature_verification.py

```diff
@@ -227,23 +232,58 @@ def _fetch_channel_signing_data(
                 f"Invalid JSON returned from {signing_data_url}/{filename}"
             )
 
-    def __call__(self, info, fn, signatures):
-        if not self.enabled or fn not in signatures:
+    def verify(self, repodata_fn: str, record: PackageRecord):
+        repodata, _ = SubdirData(
+            record.channel,
+            repodata_fn=repodata_fn,
+        ).repo_fetch.fetch_latest_parsed()
+
+        # short-circuit if no signatures are defined
+        if "signatures" not in repodata:
+            record.metadata.add(
+                f"(no signatures found for {record.channel.canonical_name})"
+            )
+            return
+        signatures = repodata["signatures"]
+
+        # short-circuit if no signature is defined for this package
+        if record.fn not in signatures:
+            record.metadata.add(f"(no signatures found for {repodata.fn})")
             return
+        signature = signatures[record.fn]
+
+        # extract metadata to be verified
+        if record.fn.endswith(CONDA_PACKAGE_EXTENSION_V1):
+            info = repodata["packages"][record.fn]
+        elif record.fn.endswith(CONDA_PACKAGE_EXTENSION_V2):
```

Comment:
This system of endswith/repodata[key] is unfortunate.

Reply:
agreed, open to suggestions